### PR TITLE
feat: add month and year pickers to DateRangePicker and RangeCalendar

### LINKED ISSUE #3089: [BUG] - showMonthAndYearPickers prop is not working in Range Calender Component
### NextUI Version

2.3.6

### Describe the bug

Calender Range component has listed `showMonthAndYearPickers` prop api , i expected it to work like it works in Calender component.

### Your Example Website or App

_No response_

### Steps to Reproduce the Bug or Issue

Just import Range Calender component and after rendering check it if we can select month/year .

### Expected behavior

Expected behaviors is just like `Calender` component which supports month and year picker for rapid selection when `showMonthAndYearPickers ` prop is used.

### Screenshots or Videos

Screenshot 2024-05-27 103452

### Operating System Version

Windows

### Browser

Edge

### LINKED ISSUE #3090: [Feature Request] Year picker for rapid selection in Date Range Picker Component
### Is your feature request related to a problem? Please describe.

Currently, Date Range Picker component doesn't support rapid selection of year. Its time consuming if I have to choose custom year. 

### Describe the solution you'd like

Just like how `showMonthAndYearPickers` works in the Calender component, it would be great if I could rapidly change to any random year in Date Range Picker component.

### Describe alternatives you've considered

NONE

### Screenshots or Videos

Screenshot 2024-05-27 111609

### PR DESCRIPTION
**Pull request by ryo-manba**
- Closes #3089 
- Closes #3090 

## 📝 Description

Added month and year pickers to DateRangePicker and RangeCalendar.

## ⛳️ Current behavior (updates)

DateRangePicker and RangeCalendar do not have month and year pickers.

## 🚀 New behavior

DateRangePicker and RangeCalendar now support month and year pickers.

## 💣 Is this a breaking change (Yes/No):

No.

## Summary by CodeRabbit

- **New Features**
  - Added month and year picker functionality to DateRangePicker and RangeCalendar components.

- **Enhancements**
  - Improved calendar components to include options for month and year selection.
  - Updated documentation to reflect new features and changes.

- **Bug Fixes**
  - Adjusted `visibleMonths` property to disable month and year pickers when set to a number greater than 1.

- **Style**
  - Added `aria-label` for better accessibility in calendar components.

### CODE REVIEW

**Comment by jrgarciadev** [comment]
@ryo-manba please adjust the range calendar docs and make clear that it only works when visible months is `1`, add examples as well to `range-calendar` and `date-range-picker` docs

**Comment by ryo-manba** [comment]
@jrgarciadev 
Sorry for the delay. I have updated the documentaton!

**Comment by Artem2001S** [comment]
@ryo-manba 
when is the pull request scheduled to be closed ?

**Comment by wingkwong** [comment]
@Artem2001S This PR is marked in 2.5.0 milestone. We'll merge this after v2.4.3.

File: .changeset/kind-cobras-travel.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@nextui-org/calendar": minor
+"@nextui-org/date-picker": minor
```
Comment (jrgarciadev) [inline .changeset/kind-cobras-travel.md:3]:
```suggestion
"@nextui-org/calendar": patch
"@nextui-org/date-picker": patch
```

**Review by jrgarciadev — APPROVED** [review]

**Pull request merged**